pkg/doc/verifiable/credential_jwt.go

```diff
@@ -0,0 +1,260 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package verifiable
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/square/go-jose/v3"
+	"github.com/square/go-jose/v3/jwt"
+)
+
+// JWTCredClaims is JWT Claims extension by Verifiable Credential.
+type JWTCredClaims struct {
+	*jwt.Claims
+
+	Credential *rawCredential `json:"vc,omitempty"`
+}
+
+// MarshalSigned marshals JWT into signed form (JWT)
+func (vcc *JWTCredClaims) MarshalSigned(signatureAlg JWTAlgorithm, privateKey interface{}, keyID string) (string, error) { //nolint:lll
+	key := jose.SigningKey{Algorithm: signatureAlg.Jose(), Key: privateKey}
+
+	var signerOpts = &jose.SignerOptions{}
+	signerOpts.WithType("JWT")
+	signerOpts.WithHeader("kid", keyID)
+
+	signer, err := jose.NewSigner(key, signerOpts)
+	if err != nil {
+		return "", fmt.Errorf("failed to create signer: %w", err)
+	}
+
+	// create an instance of Builder that uses the rsa signer
+	builder := jwt.Signed(signer).Claims(vcc)
+
+	// validate all ok, sign with the RSA key, and return a compact JWT
+	rawJWT, err := builder.CompactSerialize()
+	if err != nil {
+		return "", fmt.Errorf("failed to sign JWT: %w", err)
+	}
+
+	return rawJWT, nil
+}
+
+// MarshalUnsecured marshals JWT into unsecured JWT.
+func (vcc *JWTCredClaims) MarshalUnsecured() (string, error) {
+	headers := map[string]string{
+		"alg": "none",
+	}
+	return marshalUnsecuredJWT(headers, vcc)
+}
+
+// jwtRawCredClaims is used to get raw content of "vc" claim of JWT.
+type jwtRawCredClaims struct {
+	*jwt.Claims
+
+	Raw map[string]interface{} `json:"vc,omitempty"`
+}
+
+func (rawClaims *jwtRawCredClaims) mergeRefinedVC(raw *rawCredential) error {
+	rawVCClaims := rawClaims.Raw
+
+	rawData, err := json.Marshal(raw)
+	if err != nil {
+		return err
+	}
+
+	var rawMap map[string]interface{}
+
+	err = json.Unmarshal(rawData, &rawMap)
+	if err != nil {
+		return err
+	}
+
+	// make the merge
+	for k, v := range rawMap {
+		rawVCClaims[k] = v
+	}
+
+	return nil
+}
+
+// credJWTDecoder parses JWT claims from serialized token into rawCredential and byte array format.
+// The implementation depends on the type of serialization, e.g. signed (JWT), unsecured (plain JWT).
+type credJWTDecoder interface {
+	parseJWTClaims(rawJwt []byte) (*JWTCredClaims, error)
+	parseJWTRawClaims(rawJwt []byte) (*jwtRawCredClaims, error)
```

Comment:
The word "Raw" doesn't convey the difference between these two functions (to me).

Reply:
OK, I will find a better naming.